### PR TITLE
2229 move navigation out of delete hooks

### LINKED ISSUE #2229: Move navigation out of delete hooks
## Is your feature request related to a problem? Please describe 👀
Added in #2218. See https://github.com/openmsupply/open-msupply/pull/2218#pullrequestreview-1621146784 and https://github.com/openmsupply/open-msupply/pull/2218#issuecomment-1716674411

Also noted that navigation exists in `Insert` hooks and in `CreateOutboundFromResponse` (:

## Describe the solution you'd like 🎁
Move them out of the hooks and into the respected UI areas.

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #2229

# 👩🏻‍💻 What does this PR do? 
Refactor to move all navigation out of hooks.

# 🧪 How has/should this change been tested? 
- [ ] Ensure that you still get navigated to new page when creating Inbound, Outbound, Stocktake, Requisition and Internal Orders

### CODE REVIEW

**Review by clemens-msupply — CHANGES_REQUESTED** [review]
Looks good, when through the usages and think you covered all of them.

One nitty change request. There is a mix of async and .then which makes it more verbose than necessary...

File: client/packages/inventory/src/Stocktake/DetailView/SidePanel.tsx [hunk 1]
```diff
@@ -71,7 +75,13 @@ export const SidePanel = () => {
 
   const deleteAction = async () => {
     if (!stocktake) return;
-    await mutateAsync([stocktake]);
+    await mutateAsync([stocktake]).then(() => {
```
[the remark this answers is not in the document]
Comment (roxy-dao) [reply client/packages/inventory/src/Stocktake/DetailView/SidePanel.tsx]: [29c536e](https://github.com/msupply-foundation/open-msupply/pull/3396/commits/29c536e0c48b446a4300107cd2e3b5d29694b48b) 🙏🏻

[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/inventory/src/Stocktake/DetailView/SidePanel.tsx]: Sorry I meant for all the other changes as well...

[the remark this answers is not in the document]
Comment (roxy-dao) [reply client/packages/inventory/src/Stocktake/DetailView/SidePanel.tsx]: Done for all the deletes... I need the numbers for creation...

**Review by clemens-msupply — APPROVED** [review]

File: client/packages/inventory/src/Stocktake/DetailView/SidePanel.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/inventory/src/Stocktake/DetailView/SidePanel.tsx]: thanks!

**Pull request merged**